vault/ui_custom_messages/message.go

```diff
@@ -0,0 +1,82 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package uicustommessages
+
+import "time"
+
+var allowedMessageTypes = [2]string{
+	"banner",
+	"modal",
+}
+
+// Message is a struct that contains the fields of a particular custom message.
+type Message struct {
+	ID            string         `json:"id"`
+	Title         string         `json:"title"`
+	Message       string         `json:"message"`
+	Type          string         `json:"type"`
+	Authenticated bool           `json:"authenticated"`
+	StartTime     time.Time      `json:"start_time"`
+	EndTime       time.Time      `json:"end_time"`
+	Options       map[string]any `json:"options"`
```

Comment:
What are `Options` and `Link` used for?

Reply:
I didn't realize it when I was typing this comment, but it might make sense to add docs to these so others don't have the same question in the future